public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts

```diff
@@ -90,8 +97,13 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
 
     // Handle edit panel state
     if (typeof values.editPanel === 'string') {
-      const panel = findVizPanelByKey(this._scene, values.editPanel);
+      let panel: VizPanel | LibraryVizPanel | null;
+      panel = findVizPanelByKey(this._scene, values.editPanel);
       if (!panel) {
+        panel = findLibraryPanelByKey(this._scene, values.editPanel);
+      }
```

Comment:
Not super fond of this. But did not want findVizPanelByKey to return `VizPanel | LibraryVizPanel` and then have to handle that everywhere findVizPanelByKey is used.